src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -43,8 +44,8 @@ public MemoryCache(IOptions<MemoryCacheOptions> optionsAccessor)
             }
 
             var options = optionsAccessor.Value;
-            _entries = new Dictionary<object, CacheEntry>();
-            _entryLock = new ReaderWriterLockSlim();
+            _entries = new ConcurrentDictionary<object, CacheEntry>();
```

Comment:
Why have two fields pointing to the same object, instead of just casting when needed?